narwhal/consensus/src/consensus.rs

```diff
@@ -31,6 +35,114 @@ pub mod consensus_tests;
 /// The representation of the DAG in memory.
 pub type Dag = BTreeMap<Round, HashMap<AuthorityIdentifier, (CertificateDigest, Certificate)>>;
 
+#[derive(Default)]
+pub struct LeaderSwapTable {
+    pub good_nodes: Vec<AuthorityIdentifier>,
+    pub bad_nodes: HashSet<AuthorityIdentifier>,
+}
```

Comment:
For debugging, ideally LeaderSwapTable should store the round where it takes effect.